components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java

```diff
@@ -117,8 +135,23 @@ int getCountOfAllApplications(String tenantDomain, String username)
      * @param filter       Application name filter
      * @return int
      * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
+     * @Deprecated This is being deprecated as introducing excludeSystemPortals to exclude system portals in response.
+     * use {@link ApplicationPaginationAndSearching#getCountOfApplications(String, String, String, Boolean)}.
      */
     int getCountOfApplications(String tenantDomain, String username, String filter)
             throws IdentityApplicationManagementException;
 
+    /**
+     * Get count of all basic application information for a matching filter.
+     *
+     * @param tenantDomain          Tenant Domain.
+     * @param username              User Name.
+     * @param filter                Application name filter.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return int Count of applications.
+     * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
+     */
+    int getCountOfApplications(String tenantDomain, String username, String filter, Boolean excludeSystemPortals)
```

Comment:
Should be a default method.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b